test/suites/optional/cert-manager.robot

```diff
@@ -126,12 +173,206 @@ Apply YAML Manifest
     [Documentation]    Apply YAML manifest to the cluster
     [Arguments]    ${yaml_content}
     ${temp_file}=    Create Random Temp File    ${yaml_content}
-    ${result}=    Run With Kubeconfig    oc apply -f ${temp_file}
+    ${result}=    Oc Apply    -f ${temp_file}
     Remove File    ${temp_file}
     Should Contain    ${result}    created    msg=Failed to apply YAML manifest
     Log    Applied manifest: ${result}
 
 Setup Namespace
     [Documentation]    Setup namespace for cert-manager tests
     VAR    ${NAMESPACE}=    ${CERTS_NAMESPACE}    scope=SUITE
-    Create Namespace    ${CERTS_NAMESPACE}
+    Create Namespace If Not Exists    ${CERTS_NAMESPACE}
+
+Generate Random HostName
+    [Documentation]    Generate Random Hostname
+    ${rand}=    Generate Random String
+    ${rand}=    Convert To Lower Case    ${rand}
+    RETURN    ${rand}.api.com
+
+Add Entry To Hosts
+    [Documentation]    Add new entry to local /etc/hosts
+    [Arguments]    ${ip}    ${host}
+    VAR    ${ttt}=    ${ip}\t${host} # RF test marker\n
+    ${result}=    Run Process    sudo tee -a /etc/hosts    shell=True    stdin=${ttt}
+    Should Be Equal As Integers    ${result.rc}    0
+
+Remove Entry From Hosts
+    [Documentation]    Removes entry from local /etc/hosts
+    [Arguments]    ${host}
+    ${result}=    Run Process    sudo sed -i "/${host} # RF test marker/d" /etc/hosts    shell=True
+    Should Be Equal As Integers    ${result.rc}    0
+
+Setup Pebble Server
+    [Documentation]    Sets up a Pebble ACME server for HTTP01 testing, creates namespace if needed
+    [Arguments]    ${namespace}
+    Create Namespace If Not Exists    ${namespace}
+    ${result}=    Oc Apply    -f ${PEBBLE_DEPLOYMENT_FILE} -n ${namespace}
+
+    # Wait for Pebble deployment to be ready
+    Wait Until Keyword Succeeds    12x    10s    Check Pebble Deployment Ready    ${namespace}
+
+    VAR    ${endpoint}=    https://pebble.${namespace}.svc.cluster.local:14000/dir
+    Log    Pebble server setup successfully! (endpoint ${endpoint})
+    RETURN    ${endpoint}
+
+Create Namespace If Not Exists
+    [Documentation]    Creates a namespace if it doesn't already exist
+    [Arguments]    ${namespace}
+    ${rc}=    Run Keyword And Return Status    Oc Get    namespace ${namespace}
```

Comment:
looks like this method doesnt get correctly, it tries recreate namespace even thought its exists.
```suggestion
    ${rc}=    Run Keyword And Return Status    Oc Get    namespace    all    ${namespace}
```
BTW - I dont think you need to create NS here, there is builtin common methods for that that is called in Setup .